### ISSUE TITLE
Pulumi AI hallucinating SubnetNetworkSecurityGroupAssociation in Azure Native, the resource is only in azure classic

### ISSUE THREAD

**Issue by tusharshahrs** [pulumi/pulumi-ai#60]
### What happened?

Pulumi AI is recommending a resource that the `azure-native` provider doesn't have. `SubnetNetworkSecurityGroupAssociation` only exists in azure classic. https://www.pulumi.com/ai/answers/is6zrPx2C2CztJ5QZanLod/configuring-azure-network-security-group-associations

### Example

The following does NOT exist in [azure native](https://www.pulumi.com/registry/packages/azure-native/api-docs/network/subnet/):
`SubnetNetworkSecurityGroupAssociation`

Pulumi AI shows this:
```
const subnetNetworkSecurityGroupAssociation = new azure_native.network.SubnetNetworkSecurityGroupAssociation
```

The `SubnetNetworkSecurityGroupAssociation` exists in azure classic here:
https://www.pulumi.com/registry/packages/azure/api-docs/network/subnetnetworksecuritygroupassociation/  

### Output of `pulumi about`

n/a

### Additional context

Pulumi AI is hallucinating a resource that exist in azure classic.  This resource does NOT exist in azure-native.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by wongstein** [comment]
Early investigation: It seems Pulumi.ai is prompted with azure.network.SubnetNetworkSecurityGroupAssociation and also instructed to prefer azure_native packages, which is confusing it.

**Comment by wongstein** [comment]
Addressed and closed with commit 8dcb71e832b6908533b38dd998a1cf6ac76298dc
